work item count for an iteration (create/update/show/create-child).

pranavgore09 opened the pull request:
Related story https://github.com/fabric8io/fabric8-planner/issues/647
Allows UI to use same model as there is same response for all iteration related APIs.

Like we did in list-iterations response, need to add  WI counts in response of create/update/show/create-child iterations.
Added and modified tests around WI counts.

sbose78 commented on iteration_test.go:
```diff
@@ -146,6 +155,56 @@ func (rest *TestIterationREST) TestSuccessUpdateIteration() {
+	// add WI to this iteration and test counts in the response of update iteration API
+	wirepo := workitem.NewWorkItemRepository(rest.DB)
+	for i := 0; i < 4; i++ {
+		wirepo.Create(
```
Some success/failure assertion here would be good since you have a dependent code following this.

sbose78 commented on iteration_test.go:
```diff
@@ -146,6 +155,56 @@ func (rest *TestIterationREST) TestSuccessUpdateIteration() {
+			}, "xx")
+	}
+	svc, ctrl := rest.SecuredController()
+	_, updated := test.UpdateIterationOK(t, svc.Context, svc, ctrl, itr.ID.String(), &payload)
```
`assert`/`require` if this was a success, please?

hectorj2f commented on workitem/undoable_wi_repo.go:
```diff
@@ -125,3 +125,7 @@ func (r *UndoableWorkItemRepository) List(ctx context.Context, criteria criteria
 	return map[string]WICountsPerIteration{}, nil
 }
+
+func (r *UndoableWorkItemRepository) GetCountsForIteration(ctx context.Context, iterationId uuid.UUID) (map[string]WICountsPerIteration, error) {
```
Do we really need these functions ? We don't even use the input arguments. The just return a map... it makes no much sense to me. The same applies to the function `GetCountsPerIteration`.

hectorj2f replied:
@pranavgore09 I saw the code below in `workitem_reposity`. It makes more sense now... although is ugly. Not a blocker here

hectorj2f commented on iteration.go:
```diff
@@ -61,9 +61,11 @@ func (c *IterationController) CreateChild(ctx *app.CreateChildIterationContext)
+		wiCounts := make(map[string]workitem.WICountsPerIteration)
 		res := &app.IterationSingle{
-			Data: ConvertIteration(ctx.RequestData, &newItr),
+			Data: ConvertIteration(ctx.RequestData, &newItr, UpdateIterationsWithCounts(wiCounts)),
```
+1

hectorj2f commented:
@pranavgore09 LGTM too once you address the changes proposed by @sbose78 👍

pranavgore09 commented:
thank you @hectorj2f and @sbose78 for you time and inputs.
I have addressed all those comments here https://github.com/almighty/almighty-core/pull/815/commits/4ac7a5583469d29a24931335d64871a7d4666a24
